nca/Parsers/IstioGatewayYamlParser.py

```diff
@@ -0,0 +1,142 @@
+            self.warning(f'selector {selector} does not reference any pods in Gateway {gtw_name}. Ignoring the gateway')
+            return
+
+        gateway.peers = peers
+        self.parse_gateway_servers(gtw_name, gtw_spec, gateway)
+        self.add_gateway(gateway)
+
+    def parse_gateway_servers(self, gtw_name, gtw_spec, gateway):
+        """
+        Parses servers list in Gateway resource.
+        :param gtw_name: the gateway name
+        :param gtw_spec: the gateway spec
+        :param gateway: the parsed gateway, to include the resulting parsed servers
+        """
+        servers = gtw_spec['servers']
+
+        for i, server in enumerate(servers, start=1):
+            self.check_fields_validity(server, f'the server #{i} of the  Gateway {gtw_name}',
+                                       {'port': 1, 'bind': [0, str], 'hosts': [1, list], 'tls': 0, 'name': [0, str]})
+            port = self.parse_gateway_port(server)
+            gtw_server = Gateway.Server(port)
+            gtw_server.name = server.get('name')
+            hosts = server['hosts']
+            for host in hosts or []:
+                host_dfa = self.parse_host_value(host, gtw_spec)
+                if host_dfa:
+                    gtw_server.add_host(host_dfa)
+            if not gtw_server.hosts_dfa:
+                self.syntax_error(f'no valid hosts found for the server {gtw_server.name or i} '
+                                  f'of the Gateway {gtw_name}')
+            gateway.add_server(gtw_server)
+
+    def parse_gateway_port(self, server):
+        port = server['port']
+        self.check_fields_validity(port, f'the port of the server {server}',
+                                   {'number': [1, int], 'protocol': [1, str], 'name': [1, str], 'targetPort': [3, str]})
+        number = port['number']
+        protocol = port['protocol']
+        name = port['name']
+        return Gateway.Server.GatewayPort(number, protocol, name)
+
+    @staticmethod
+    def get_egress_gtw_pods(pods):
+        """
+        Heuristically identifying egress gateway pods out of given pods.
+        Detection of egress gateway pods is based on having "egress" substring in pod names or labels.
+        :param PeerSet pods: a given set of pods
+        :return: a set of egress gateway pods
+        """
+        result = PeerSet()
+        look_for = "egress"
+        for peer in pods:
+            # first, try to look in peer's name
+            if look_for in peer.name:
+                result.add(peer)
+                continue
+            # try to look in peer's labels
+            found = [val for val in list(peer.labels.values()) if look_for in val]
+            if found:
+                result.add(peer)
+                continue
+            # still not found? try to look in extra_labels
+            found = [val for val in list(peer.extra_labels.values()) if look_for in val]
+            if found:
+                result.add(peer)
+        return result
```

Comment:
Is it possible to identify the direction of the rules instead of relying on this heuristics, for inferring "egress flow" in gateway analysis?

Reply:
According to our discussion, supporting only standard istio ingress/egress gateways, which are identified by `'istio: ingressgateway'` and `'istio: egressgateway'` selectors, and egress gateway destination is identified by `'istio-egressgateway.istio-system.svc.cluster.local'` host.
Updated relevant tests and expected results correspondingly.